<a href="https://colab.research.google.com/github/JonHaz/Capstone-2-LVMH-Sales-in-Iowa/blob/main/LVMH_Capstone_Two_Data_Wrangling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

LVMH Moët Hennessy Louis Vuitton, commonly known as LVMH, is a French Holding multinational corporation and conglomerate specializing in luxury goods, headquartered in Paris, France. Is brand that steadily becoming a household name, but in terms of their footprint in Iowa what can we learn from their Wine and Spirit sales within the state.

In my role as a Product Analyst I often use BigQuery and other Cloud Datawarehouses so I thought it be interesting to complete this project entirely in the cloud.

In [1]:
# I am establishing a connection to my Sandbox within BigQuery so that I can run SQL queries to interact with the data here within Google Collab
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


In [4]:
# Lets see what data we are working with
%%bigquery --project storied-sound-253213 
SELECT * 
FROM `bigquery-public-data.iowa_liquor_sales.sales`
LIMIT 10

invoice_and_item_number        date  ... volume_sold_liters volume_sold_gallons
0         INV-25006100046  2020-02-04  ...               48.0               12.68
1         INV-25048000012  2020-02-06  ...               72.0               19.02
2         INV-30846500020  2020-10-07  ...               18.0                4.75
3         INV-30951200042  2020-10-12  ...               22.5                5.94
4         INV-30910900037  2020-10-09  ...               90.0               23.77
5         INV-30953700158  2020-10-12  ...                0.2                0.05
6            S29656700013  2015-12-15  ...                4.0                1.06
7         INV-23092300034  2019-11-07  ...                0.6                0.15
8            S15040800017  2013-10-09  ...                4.0                1.06
9            S26173800003  2015-06-18  ...              156.0               41.21

[10 rows x 24 columns]

In [3]:
#Seeing that I am primarily interested in the sales from LVMH for this project I am going to run a query to determine what is the vendor name

%%bigquery --project storied-sound-253213 
SELECT DISTINCT vendor_name, item_description
FROM `bigquery-public-data.iowa_liquor_sales.sales`
WHERE vendor_name LIKE '%HENNESSY%'

vendor_name                      item_description
0    MOET HENNESSY USA              Glenmorangie Taster Pack
1    MOET HENNESSY USA             Glenmorangie Pioneer Pack
2    MOET HENNESSY USA                  Glenmorangie Lasanta
3    MOET HENNESSY USA                     Glenmorangie 10YR
4    MOET HENNESSY USA                          Ardbeg An Oa
..                 ...                                   ...
116  MOET HENNESSY USA        Belvedere Pure w/Martini Glass
117  MOET HENNESSY USA             Hennessy Paradis Imperial
118  MOET HENNESSY USA    HA Glenmorangie Grand Vintage 1991
119  MOET HENNESSY USA       Hennessy Paradis Imperial 750ML
120  MOET HENNESSY USA  Hennessy VSOP 2020 EOY Manhattan VAP

[121 rows x 2 columns]

I recognize these brands and the vendor name looks like a match. So what I will do is load all of the data related to the vendor MOET HENNESSY USA into a Pandas DataFrame

In [5]:
#Google collab load Pandas by default but typically I would 
#import pandas as pd
# Save output in a variable `df`
%%bigquery --project storied-sound-253213 df
SELECT *
FROM `bigquery-public-data.iowa_liquor_sales.sales`
WHERE vendor_name LIKE '%HENNESSY%'

In [6]:
#Lets inspect our new dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 175379 entries, 0 to 175378
Data columns (total 24 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   invoice_and_item_number  175379 non-null  object 
 1   date                     175379 non-null  object 
 2   store_number             175379 non-null  object 
 3   store_name               175379 non-null  object 
 4   address                  173783 non-null  object 
 5   city                     173783 non-null  object 
 6   zip_code                 173783 non-null  object 
 7   store_location           158121 non-null  object 
 8   county_number            173783 non-null  object 
 9   county                   173783 non-null  object 
 10  category                 174978 non-null  object 
 11  category_name            174978 non-null  object 
 12  vendor_number            175379 non-null  object 
 13  vendor_name              175379 non-null  object 
 14  item

It looks like there are 175,379 entries and none of the field are null!

But it appears that not all of the columns have the correct datatype so we would want to fix this.

In [7]:
df.head()

invoice_and_item_number        date  ... volume_sold_liters volume_sold_gallons
0         INV-30752900191  2020-10-05  ...               13.5                3.56
1         INV-20918000010  2019-07-30  ...               13.5                3.56
2         INV-10472100006  2018-02-19  ...               22.5                5.94
3         INV-07182900001  2017-09-12  ...               27.0                7.13
4         INV-06905200006  2017-08-28  ...                4.8                1.27

[5 rows x 24 columns]

In [12]:
#in order to use the pd shorthand for pandas it seems i do need to 
#import pandas as pd
#lesson learned

import pandas as pd
df['date'] = pd.to_datetime(df['date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 175379 entries, 0 to 175378
Data columns (total 24 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   invoice_and_item_number  175379 non-null  object        
 1   date                     175379 non-null  datetime64[ns]
 2   store_number             175379 non-null  object        
 3   store_name               175379 non-null  object        
 4   address                  173783 non-null  object        
 5   city                     173783 non-null  object        
 6   zip_code                 173783 non-null  object        
 7   store_location           158121 non-null  object        
 8   county_number            173783 non-null  object        
 9   county                   173783 non-null  object        
 10  category                 174978 non-null  object        
 11  category_name            174978 non-null  object        
 12  vendor_number   

Before going forward one thing i want to verify is what field can be used as an index

In [16]:
df.nunique()

invoice_and_item_number    175379
date                         1256
store_number                 1952
store_name                   2027
address                      1898
city                          397
zip_code                      434
store_location               1678
county_number                  99
county                        129
category                       10
category_name                  10
vendor_number                   1
vendor_name                     1
item_number                   120
item_description              121
pack                            9
bottle_volume_ml               11
state_bottle_cost              86
state_bottle_retail            86
bottles_sold                   91
sale_dollars                 1112
volume_sold_liters            176
volume_sold_gallons           222
dtype: int64

Okay so this tells me quite a lot about my dataframe such as

*   df['invoice_and_item_number'] are my unique values
*   There are 10 categories, 9 differenct packages and 11 bottle sizes

At a glance my data looks nice, clean and tidy.